lib/stm32/f3/adc.c

```diff
@@ -282,14 +282,11 @@ void adc_enable_analog_watchdog_on_all_channels(uint32_t adc)
 void adc_enable_analog_watchdog_on_selected_channel(uint32_t adc,
 						    uint8_t channel)
 {
-	uint32_t reg32;
+	ADC_CFGR1(adc) = (ADC_CFGR1(adc) & ~ADC_CFGR1_AWD1CH) |
+			  ADC_CFGR1_AWD1CH_VAL(channel);
 
-	reg32 = (ADC_CFGR1(adc) & ~ADC_CFGR1_AWD1CH); /* Clear bit [4:0]. */
-	if (channel < 18) {
-		reg32 |= channel;
-	}
-	ADC_CFGR1(adc) = reg32;
-	ADC_CFGR1(adc) |= ADC_CFGR1_AWD1SGL;
+	// enable watchdog on a single channel (as opposed to all channels)
```

Comment:
no c++ comments please.  Further, I don't get this one, it says "unify with f0" but it's still in the f3 code?  is this now the same function in two files?  Is it now just a function that should go into adc_common_v2?

Reply:
Right, I see that that is the case.  I've dropped the c++ comment, and am merging this, and will file a "easy pickings" issue for someone who wants to extract this code, there's no need for it to block this commit